packages/trader/src/Modules/SmartChart/Components/all-markers.jsx

```diff
@@ -233,6 +240,72 @@ const drawAccuBarrierRange = ({
     ctx.restore();
 };
 
+const draw_reset_barrier = ({
+    ctx,
+    start_left,
+    top,
+    bottom,
+    stroke_color,
+    has_persistent_borders = false,
+    barrier_gradient,
+    reset_barrier,
+    scale,
+}) => {
+    ctx.save();
+    const end_left = ctx.canvas.offsetWidth - ctx.canvas.parentElement.stx.panels.chart.yaxisTotalWidthRight;
+    const end_top = ctx.canvas.offsetHeight - ctx.canvas.parentElement.stx.xaxisHeight;
+    const is_top_visible = top < end_top && (top >= 0 || !has_persistent_borders);
+    const is_bottom_visible = bottom < end_top;
+    // using 2 instead of 0 to distance the top barrier line from the top of the chart and make it clearly visible in C.Details:
+    const persistent_top = top < 0 && has_persistent_borders ? 2 : end_top;
+    const displayed_top = is_top_visible ? top : persistent_top;
+    const displayed_bottom = is_bottom_visible ? bottom : end_top;
+    const is_start_left_visible = start_left < end_left;
+
+    if (!is_start_left_visible) return;
+    ctx.lineWidth = 1;
+    ctx.strokeStyle = stroke_color;
+    ctx.setLineDash([]);
+    ctx.textAlign = 'right';
+
+    if ((is_top_visible || has_persistent_borders) && barrier_gradient === top) {
+        ctx.fillStyle = stroke_color;
+        ctx.beginPath();
+        ctx.setLineDash([2, 3]);
+        ctx.moveTo(start_left + 1.5 * scale, reset_barrier);
+        ctx.lineTo(end_left, reset_barrier);
+        ctx.fill();
+        ctx.stroke();
+    }
+
+    if ((is_bottom_visible || has_persistent_borders) && barrier_gradient === bottom) {
+        ctx.fillStyle = stroke_color;
+        ctx.beginPath();
+        ctx.setLineDash([2, 3]);
+        ctx.moveTo(start_left + 1.5 * scale, reset_barrier);
+        ctx.lineTo(end_left, reset_barrier);
+        ctx.fill();
+        ctx.stroke();
+    }
+    let gradient;
+    //This case is for Reset Call
+    if (barrier_gradient === bottom) {
+        gradient = ctx.createLinearGradient(start_left, displayed_bottom, start_left, displayed_top);
+        gradient.addColorStop(0.02, 'rgba(75, 180, 179, 0.16)');
+        gradient.addColorStop(0.98, 'rgba(75, 180, 179, 0.00)');
+    }
+    //This case is for Reset Put
+    if (barrier_gradient === top) {
+        gradient = ctx.createLinearGradient(start_left, displayed_top, start_left, displayed_bottom);
+        gradient.addColorStop(0.02, 'rgba(75, 180, 179, 0.16)');
+        gradient.addColorStop(0.98, 'rgba(75, 180, 179, 0.00)');
+    }
+
+    ctx.fillStyle = gradient;
```

Comment:
if this function is 99% the same as `drawAccuBarrierRange()`, it's better to add your changes to it and reuse it :) it can be named `drawBarrierRange` in this case

Reply:
5 month ago I was asking if I should reuse your function, you reply no (

Reply:
+ actually they are not 99%